.gitignore

```diff
@@ -0,0 +1,16 @@
+*.suo
+*.[Cc]ache
+*.dll
+*.exe
+[Dd]ebug/
+[Dd]ebugPublic/
+[Rr]elease/
+[Rr]eleases/
+x64/
+x86/
+build/
+bld/
+[Bb]in/
+[Oo]bj/
+obj/
+.[Cc]onfig
```

Comment:
@Zack-TT Please trim this list down.  If you ignore a directory, everything inside it will  be ignored as well, so you can remove a lot of these rules.  One example is the Debug/Release dirs from the build should not exist outside bin and obj, so ignoring only bin and obj should be enough.